src/quill-editor.component.ts

```diff
@@ -151,43 +177,51 @@ export class QuillEditorComponent implements AfterViewInit, ControlValueAccessor
     this.onEditorCreated.emit(this.quillEditor);
 
     // mark model as touched if editor lost focus
-    this.quillEditor.on('selection-change', (range: any, oldRange: any, source: string) => {
-      this.zone.run(() => {
-        this.onSelectionChanged.emit({
-          editor: this.quillEditor,
-          range: range,
-          oldRange: oldRange,
-          source: source
+    this.quillEditor.on(
+      'selection-change',
+      (range: any, oldRange: any, source: string) => {
+        this.zone.run(() => {
+          this.onSelectionChanged.emit({
+            editor: this.quillEditor,
+            range: range,
+            oldRange: oldRange,
+            source: source
+          });
+
+          if (!range) {
+            this.onModelTouched();
+          }
         });
-
-        if (!range) {
-          this.onModelTouched();
-        }
-      });
-    });
+      }
+    );
 
     // update model if text changes
-    this.quillEditor.on('text-change', (delta: any, oldDelta: any, source: string) => {
-      let html: (string | null) = this.editorElem.children[0].innerHTML;
-      const text = this.quillEditor.getText();
+    this.quillEditor.on(
+      'text-change',
+      (delta: any, oldDelta: any, source: string) => {
+        const text = this.quillEditor.getText();
 
-      if (html === '<p><br></p>') {
+        let html: string | null = this.editorElem.children[0].innerHTML;
```

Comment:
you can replace this part with your getter, right? so you can call the getter one time and reuse the return value for the output and onModelChange

Reply:
not really because i like that you receive the html with the event "onContentChanged". if i would use the valueGetter i'm not sure what i'm getting back. that's why the html must be extracted separately.